Add compiling with MinGW to install documentation

This is working and should be documented in the install instructions.
```
mkdir build
cd build
cmake -G "MinGW Makefiles" -DCMAKE_INSTALL_PREFIX=path_to_prefix ..
mingw32-make
mingw32-make install
```

Update:
`cmake`, `make `and `make install` for xtl and xtensor works.

However by just including `#include <xtensor/xarray.hpp>` I get 3962 compiler errors.
One of which being: `xtype_traits.hpp:54: error: 'decay_t' is not a member of 'std'`

The used compiler is:
`gcc (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 7.3.0`

--- a/docs/source/installation.rst
+++ b/docs/source/installation.rst
@@ -94,6 +94,16 @@ this folder contains ``include`` and ``lib`` subfolders.
 
 See the :doc:`build-options` section for more details about cmake options.
 
+Although not officially supported, ``xtensor`` can be installed with MinGW:
+
+.. code::
+
+    mkdir build
+    cd build
+    cmake -G "MinGW Makefiles" -DCMAKE_INSTALL_PREFIX=path_to_prefix ..
+    mingw32-make
+    mingw32-make install
+
 Including xtensor in your project
 ---------------------------------
 

